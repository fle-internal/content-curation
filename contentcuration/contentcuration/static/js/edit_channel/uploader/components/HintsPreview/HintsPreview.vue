<template>
  <div>
    <div v-if="!hints || !hints.length">
      No hints yet
    </div>

    <div v-else>
      <span
        class="toggle"
        data-test="toggle"
        @click="toggle"
      >
        <span>{{ toggleLabel }}</span>

        <span class="icon">
          <v-icon v-if="isOpen">arrow_drop_down</v-icon>
          <v-icon v-else>arrow_drop_up</v-icon>
        </span>
      </span>

      <VList v-if="isOpen">
        <VListTile
          v-for="(hint, hintIdx) in hints"
          :key="hintIdx"
        >
          <VFlex xs1>
            {{ hint.order }}
          </VFlex>
          <VFlex xs2>
            {{ hint.hint }}
          </VFlex>
        </VListTile>
      </VList>
    </div>
  </div>
</template>

<script>

  export default {
    name: 'HintsPreview',
    props: {
      hints: {
        type: Array,
      },
    },
    data() {
      return {
        isOpen: false,
      };
    },
    computed: {
      toggleLabel() {
        if (this.isOpen) {
          return 'Hide hints';
        }

        if (this.hintsCount === 1) {
          return 'Show 1 hint';
        }

        return `Show ${this.hintsCount} hints`;
      },
      hintsCount() {
        if (!this.hints) {
          return 0;
        }

        return this.hints.length;
      },
    },
    methods: {
      toggle() {
        this.isOpen = !this.isOpen;
      },
    },
  };

</script>

<style lang="less" scoped>
  .toggle {
    text-decoration: underline;
    cursor: pointer;

    .icon {
      position: relative;
      top: 3px;
    }
  }
</style>
