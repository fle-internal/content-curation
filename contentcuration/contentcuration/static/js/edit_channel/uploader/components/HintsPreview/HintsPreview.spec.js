import { shallowMount, mount } from '@vue/test-utils';

import HintsPreview from '../HintsPreview/HintsPreview.vue';

const clickToggle = wrapper => {
  wrapper.find('[data-test=toggle]').trigger('click');
};

describe('HintsPreview', () => {
  let wrapper;

  it('smoke test', () => {
    const wrapper = shallowMount(HintsPreview);

    expect(wrapper.isVueInstance()).toBe(true);
  });

  describe('with no hints', () => {
    beforeEach(() => {
      wrapper = mount(HintsPreview, {
        propsData: {
          hints: [],
        },
      });
    });

    it('renders placeholder text only', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('with one hint', () => {
    beforeEach(() => {
      wrapper = mount(HintsPreview, {
        propsData: {
          hints: [{ order: 1, hint: 'Hint 1' }],
        },
      });
    });

    it('renders correct toggle label', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('with more hints', () => {
    beforeEach(() => {
      wrapper = mount(HintsPreview, {
        propsData: {
          hints: [
            { order: 1, hint: 'Hint 1' },
            { order: 2, hint: 'Hint 2' },
            { order: 3, hint: 'Hint 3' },
          ],
        },
      });
    });

    it('renders correct toggle label', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('', () => {
    beforeEach(() => {
      wrapper = mount(HintsPreview, {
        propsData: {
          hints: [
            { order: 1, hint: 'Hint 1' },
            { order: 2, hint: 'Hint 2' },
            { order: 3, hint: 'Hint 3' },
          ],
        },
      });
    });

    describe('on toggle click', () => {
      beforeEach(() => {
        clickToggle(wrapper);
      });

      it('renders correct toggle label and displays hints', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });
    });

    describe('on second toggle click', () => {
      beforeEach(() => {
        clickToggle(wrapper);
        clickToggle(wrapper);
      });

      it('hides hints', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });
    });
  });
});
