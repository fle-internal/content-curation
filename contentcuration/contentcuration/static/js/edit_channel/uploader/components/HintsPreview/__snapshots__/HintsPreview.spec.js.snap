// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintsPreview  on second toggle click hides hints 1`] = `
<div>
  <div><span data-test="toggle" class="toggle"><span>Show 3 hints</span> <span class="icon"><i aria-hidden="true" class="v-icon material-icons theme--light">arrow_drop_up</i></span></span>
    <!---->
  </div>
</div>
`;

exports[`HintsPreview  on toggle click renders correct toggle label and displays hints 1`] = `
<div>
  <div><span data-test="toggle" class="toggle"><span>Hide hints</span> <span class="icon"><i aria-hidden="true" class="v-icon material-icons theme--light">arrow_drop_down</i></span></span>
    <div role="list" class="v-list theme--light">
      <div role="listitem">
        <div class="v-list__tile theme--light">
          <div class="flex xs1">
            1
          </div>
          <div class="flex xs2">
            Hint 1
          </div>
        </div>
      </div>
      <div role="listitem">
        <div class="v-list__tile theme--light">
          <div class="flex xs1">
            2
          </div>
          <div class="flex xs2">
            Hint 2
          </div>
        </div>
      </div>
      <div role="listitem">
        <div class="v-list__tile theme--light">
          <div class="flex xs1">
            3
          </div>
          <div class="flex xs2">
            Hint 3
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HintsPreview with more hints renders correct toggle label 1`] = `
<div>
  <div><span data-test="toggle" class="toggle"><span>Show 3 hints</span> <span class="icon"><i aria-hidden="true" class="v-icon material-icons theme--light">arrow_drop_up</i></span></span>
    <!---->
  </div>
</div>
`;

exports[`HintsPreview with no hints renders placeholder text only 1`] = `
<div>
  <div>
    No hints yet
  </div>
</div>
`;

exports[`HintsPreview with one hint renders correct toggle label 1`] = `
<div>
  <div><span data-test="toggle" class="toggle"><span>Show 1 hint</span> <span class="icon"><i aria-hidden="true" class="v-icon material-icons theme--light">arrow_drop_up</i></span></span>
    <!---->
  </div>
</div>
`;
